mopidy/core/library.py

```diff
@@ -136,9 +139,16 @@ def get_distinct(self, field, query=None):
         validation.check_choice(field, validation.DISTINCT_FIELDS)
         query is None or validation.check_query(query)  # TODO: normalize?
 
+        compat_field = {"track_name": "track"}.get(field, field)
+        if compat_field != field:
```

Comment:
Looking at this again, does it make sense? Shouldn't it only log a warning if `field == "track"` and shouldn't the warning be directed at the frontend which called using the old parameter value, rather than the backend which has done nothing wrong?

Reply:
Yeah, to me it looks like this will only warn if `field` is `track_name`, which is the new keyword argument you're supposed to use. Will it be correct if you change it around to:

```python
compat_field = {"track": "track_name"}.get(field, field)
```

E.g. `track` from the frontend is converted to `track_name` before it is passed to the backend?

This doesn't blame the backend, does it? If you run the test suite of a frontend using the old keyword argument with warnings converted to errors, the test suite should fail with an exeption and a traceback showing where in the frontend it called the Core API with a deprecated keyword argument.

Reply:
My thinking was that although the correct field is now going to be `track_name`, this is a breaking API change so I can't actually use `track_name` with backends yet, I have to convert it to the old field value (`track`) for the time being, but then also emit a deprecation warning so that backends get a chance to be upgraded to support the new value. Am I overthinking this?

Reply:
I think you're right. As is, with `if compat_field != field` changed to `if field == "track"` sounds like the right way. That would:

- Accept `track_name` from frontends without complaining. Translate to `track` for backends, to keep compat.
- Accept `track` from frontends and warn about it being deprecated. Pass it unchanged to the backends.